src/main/io/serial.c

```diff
@@ -421,3 +423,45 @@ void evaluateOtherData(serialPort_t *serialPort, uint8_t receivedChar)
         systemResetToBootloader();
     }
 }
+
+/*
+A low-level interface for implementing serial passthrough. This is used for
+example by gps to do serial passthrough while also consuming GPS data and
+updating the display. It is used below by the high-level `serialPassthrough()`
+function to move data between two serial ports.
+
+Read data from `left` and writes it to `right`
+Returns data (if any) to caller for further processing.
+*/
+uint8_t serialPassthroughStep(serialPort_t *left, serialPort_t *right)
+{
+    uint8_t c = 0;
+    if (serialRxBytesWaiting(left)) {
+        c = serialRead(left);
+        serialWrite(right, c);
+    }
+    return c;
+}
+
+/*
+A high-level serial passthrough implementation. Used by cli to start an
+arbitrary serial passthrough "proxy". Manages LEDs while using the low-level
+`serialPassthroughStep()` to shuffle data both ways.
+*/
+void serialPassthrough(serialPort_t *left, serialPort_t *right)
```

Comment:
You may define this function as 
```
typedef void serialConsumer(uint8_t);
static void nopConsumer(uint8_t) {}
void serialPassthrough(serialPort_t *left, serialPort_t *right, serialConsumer* leftC, serialConsumer* rightC)
{
     waitForSerialPortToFinishTransmitting(left);
     waitForSerialPortToFinishTransmitting(right);
 
     if(!leftC) 
       leftC = &nopConsumer;
     if(!rightC) 
       rightC = &nopConsumer;

     LED0_OFF;
     LED1_OFF;
 
     while(1) {
        if (serialRxBytesWaiting(left)) {
          LED0_ON;
          uint8_t c = serialRead(left);
          serialWrite(right, c);
          leftC(c);
          LED0_OFF;
        }
         if (serialRxBytesWaiting(right)) {
          LED0_ON;
          uint8_t c = serialRead(right);
          serialWrite(left, c);
          rightC(c);
          LED0_OFF;
        }
     } 
}
```
This way the code should handle generic passthrough and gps.

Also it is probably good idea to light led only when char is copied - human eye is very good at detection extremely low duty cycle, `serialRxBytesWaiting()` overhead may be high enough for led to be slightly glowing.

Also it would be great to implement mechanism to end passthrough without reset - "+++" with guard time was used in (good old) modem days : https://en.wikipedia.org/wiki/Escape_sequence#Modem_control